src/mslice/workspace/helperfunctions.py

```diff
@@ -116,3 +115,60 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         if self.workspace:
             self.workspace.remove_saved_attributes()
         return True
+
+
+class WorkspaceNameHandler:
```

Comment:
I'm surprised there are so many options! I don't think it's ideal to have a separate function for each combination of functions. It's plausible in the future we have to add more, and therefore the number of combination functions will increase exponentially.

I wonder if it's a good idea to instead have a `WorkspaceNameOptions` class, that has a constructor that default sets everything to false via named arguments.

We could then do a combination of functions by doing:
`WorkspaceNameHandler.set_workspace_options(workspace.name, WorkSpaceOptions(hidden_from_mslice=True, hidden_from_ADS=True))`

You would then only need individual functions in the `WorkspaceNameHandler` class, and these would not have to be exposed.

Perhaps when checking we could then do:
`WorkspaceNameHandler.assert_workspace_options(workspace, WorkspaceOptions(hidden_from_mslice=True, hidden_from_ADS=True)`

Reply:
I have now implemented these suggestions, but instead of creating and entirely separate `WorkspaceOptions` class, I preferred to opt for two methods `get_name` and `assert_name` that mimicks this. So in this case one would do `WorkspaceNameHandler(workspace.name).get_name(hidden_from_mslice=True, hidden_from_ADS=True` and `WorkspaceNameHandler(workspace.name).assert_name(hidden_from_mslice=True, hiddden_from_ADS=True)`.